sql/core/src/test/resources/sql-tests/inputs/udf/pgSQL/udf-join.sql

```diff
@@ -0,0 +1,2081 @@
+-- select a.idv, b.idv from tidv a, tidv b where a.idv = b.idv;
+
+-- set enable_mergejoin = 0;
+
+-- explain (costs off)
+-- select a.idv, b.idv from tidv a, tidv b where a.idv = b.idv;
+
+-- rollback;
+
+--
+-- test NULL behavior of whole-row Vars, per bug #5025
+--
+select udf(t1.q2), udf(count(t2.*))
+from int8_tbl t1 left join int8_tbl t2 on (udf(udf(t1.q2)) = t2.q1)
+group by udf(t1.q2) order by 1;
+
+select udf(udf(t1.q2)), udf(count(t2.*))
+from int8_tbl t1 left join (select * from int8_tbl) t2 on (udf(udf(t1.q2)) = udf(t2.q1))
+group by udf(udf(t1.q2)) order by 1;
+
+-- [SPARK-28330] Enhance query limit
+-- select t1.q2, count(t2.*)
+-- from int8_tbl t1 left join (select * from int8_tbl offset 0) t2 on (t1.q2 = t2.q1)
+-- group by t1.q2 order by 1;
+
+select udf(t1.q2) as q2, udf(udf(count(t2.*)))
+from int8_tbl t1 left join
+  (select udf(q1) as q1, case when q2=1 then 1 else q2 end as q2 from int8_tbl) t2
+  on (udf(t1.q2) = udf(t2.q1))
+group by t1.q2 order by 1;
+
+--
+-- test incorrect failure to NULL pulled-up subexpressions
+--
+-- begin;
+create or replace temporary view a as select * from
+  (values ('p'), ('q'))
+  as v(code);
+create or replace temporary view b as select * from
+  (values ('p', 1), ('p', 2))
+  as v(a, num);
+create or replace temporary view c as select * from
+  (values ('A', 'p'), ('B', 'q'), ('C', null))
+  as v(name, a);
+
+select udf(c.name), udf(ss.code), udf(ss.b_cnt), udf(ss.const)
+from c left join
+  (select a.code, coalesce(b_grp.cnt, 0) as b_cnt, -1 as const
+   from a left join
+     (select udf(count(1)) as cnt, b.a as a from b group by b.a) as b_grp
+     on udf(a.code) = udf(udf(b_grp.a))
+  ) as ss
+  on (udf(udf(c.a)) = udf(ss.code))
+order by c.name;
+
+-- rollback;
+
+--
+-- test incorrect handling of placeholders that only appear in targetlists,
+-- per bug #6154
+--
+SELECT * FROM
+( SELECT 1 as key1 ) sub1
+LEFT JOIN
+( SELECT sub3.key3, sub4.value2, COALESCE(sub4.value2, 66) as value3 FROM
+    ( SELECT 1 as key3 ) sub3
+    LEFT JOIN
+    ( SELECT udf(sub5.key5) as key5, udf(udf(COALESCE(sub6.value1, 1))) as value2 FROM
+        ( SELECT 1 as key5 ) sub5
+        LEFT JOIN
+        ( SELECT 2 as key6, 42 as value1 ) sub6
+        ON sub5.key5 = udf(sub6.key6)
+    ) sub4
+    ON udf(sub4.key5) = sub3.key3
+) sub2
+ON udf(udf(sub1.key1)) = udf(udf(sub2.key3));
+
+-- test the path using join aliases, too
+SELECT * FROM
+( SELECT 1 as key1 ) sub1
+LEFT JOIN
+( SELECT udf(sub3.key3) as key3, udf(value2), udf(COALESCE(value2, 66)) as value3 FROM
+    ( SELECT 1 as key3 ) sub3
+    LEFT JOIN
+    ( SELECT sub5.key5, COALESCE(sub6.value1, 1) as value2 FROM
+        ( SELECT 1 as key5 ) sub5
+        LEFT JOIN
+        ( SELECT 2 as key6, 42 as value1 ) sub6
+        ON udf(udf(sub5.key5)) = sub6.key6
+    ) sub4
+    ON sub4.key5 = sub3.key3
+) sub2
+ON sub1.key1 = udf(udf(sub2.key3));
+
+--
+-- test case where a PlaceHolderVar is used as a nestloop parameter
+--
+
+-- EXPLAIN (COSTS OFF)
```

Comment:
This is from join.sql, but looks it is pgsql specific stuff? Does it make sense to keep it? @HyukjinKwon

Reply:
Yea, actually there are multiple instances like that IIRC. Let's just keep it for simplicity for now.